keras/backend/tensorflow_backend.py

```diff
@@ -1828,6 +1828,24 @@ def less_equal(x, y):
     return tf.less_equal(x, y)
 
 
+def where(condition, x, y):
+    """Returns elements, either from x or y, depending on condition.
+       If only condition is given, \
+       (suppose we have transfer condition from Tensor to `numpy.ndarray`)
+       Return `tf.convert_to_tensor(condition.nonzero())`.
+
+    # Arguments
+        condition: Tensor or variable.
+        x: Tensor or variable.
+        y: Tensor or variable.
+
+    # Returns
+        A tensor.
+    """
+    condition = tf.cast(condition, dtype=tf.bool)
```

Comment:
Make the cast conditional on `condition` being the wrong dtype

Reply:
(is the cast really necessary? does the statement below crash without the cast?)

Reply:
Yep, if i don't add this explicit cast,  The Travis CI build gonna failed.